### ISSUE TITLE
Update browser types, adding channel and making names consistent

### ISSUE THREAD

**Issue by flotwig** [cypress-io/cypress#6243]
### Current behavior:

Browsers are defined like this:

```ts
export type Browser = {
  /** short browser name */
  name: BrowserName
  family: BrowserFamily
  /** ... **/
}
```

...which isn't well-suited to filtering out browsers based on behavior.

### Desired behavior:

Browsers are defined like this:

```ts
export type Browser = {
  /** short browser name */
  name: BrowserName
  /** family is used to describe the underlying type of the browsr */
  family: BrowserFamily
  /** channel describes the stability of this browser **/
  channel: BrowserChannel
  /** ... **/
}
```

So this:

https://github.com/cypress-io/cypress/blob/c15e8e1473b2787e0402e40811baa6a68c640650/packages/launcher/lib/browsers.ts#L8-L60

Would become this:

```ts
export const browsers: Browser[] = [
  {
    name: 'chrome',
    family: 'chromium',
    channel: 'stable',
    displayName: 'Chrome',
    versionRegex: /Google Chrome (\S+)/,
    profile: true,
    binary: ['google-chrome', 'chrome', 'google-chrome-stable'],
  },
  {
    name: 'chromium',
    family: 'chromium',
    channel: 'dev',
    displayName: 'Chromium',
    versionRegex: /Chromium (\S+)/,
    profile: true,
    binary: ['chromium-browser', 'chromium'],
  },
  {
    name: 'chrome',
    family: 'chromium',
    channel: 'canary',
    displayName: 'Canary',
    versionRegex: /Google Chrome Canary (\S+)/,
    profile: true,
    binary: 'google-chrome-canary',
  },
  {
    name: 'firefox',
    family: 'firefox',
    channel: 'stable',
    displayName: 'Firefox',
    info: firefoxInfo,
    versionRegex: /Firefox (\S+)/,
    profile: true,
    binary: 'firefox',
  },
  {
    name: 'firefox',
    family: 'firefox',
    channel: 'dev',
    displayName: 'Firefox Developer Edition',
    info: firefoxInfo,
    versionRegex: /Firefox Developer Edition (\S+)/,
    profile: true,
    binary: 'firefox-developer-edition',
  },
  {
    name: 'firefox',
    family: 'firefox',
    channel: 'nightly',
    displayName: 'Firefox Nightly',
    info: firefoxInfo,
    versionRegex: /Firefox Nightly (\S+)/,
    profile: true,
    binary: 'firefox-nightly',
  },
]
```

Microsoft Edge #5433 would look like this:

```ts
  {
    name: 'edge',
    family: 'chromium',
    channel: 'stable',
    displayName: 'Edge',
    versionRegex: /** ... **/,
    profile: /** ... **/,
    binary: /** ... **/,
  },
```

`electron` would it's family be set to `chromium`, and existing code will be updated to assert on name. Since it needs to be treated differently from normal Chromium-based browsers, but `isChromiumBased` would still be true.

This has the advantage of making filtering easy to do:

```js
// not a real API
Cypress.isBrowser({ name: 'edge' })
Cypress.isBrowser({ family: 'firefox' })
Cypress.isBrowser("foo") // same as { name: 'foo' }
```

**Comment by flotwig** [comment]
As part of this, the logic for selecting browsers to be launched will be updated.

* `--browser browserName` launches browserName with channel = `stable`
* `--browser browserName:channel` launches browserName with channel = `channel`
* `--browser canary` has been aliased to `chrome:canary` for backwards compatibility

**Comment by flotwig** [comment]
Weird, GitHub didn't pick up on the "closes" from the PR...

**Comment by flotwig** [comment]
Released in [`4.0.0`](https://on.cypress.io/changelog#4-0-0).
